src/Microsoft.Health.Fhir.Azure/IntegrationDataStore/AzureConnectionStringClientInitializerV2.cs

```diff
@@ -0,0 +1,66 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Azure.Storage;
+using Microsoft.Azure.Storage.Blob;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Fhir.Core.Configs;
+using Microsoft.Health.Fhir.Core.Features.Operations;
+
+namespace Microsoft.Health.Fhir.Azure.IntegrationDataStore
+{
+    public class AzureConnectionStringClientInitializerV2 : IIntegrationDataStoreClientInitilizer<CloudBlobClient>
+    {
+        private readonly IntegrationDataStoreConfiguration _integrationDataStoreConfiguration;
+        private readonly ILogger<AzureConnectionStringClientInitializerV2> _logger;
+
+        public AzureConnectionStringClientInitializerV2(IOptions<IntegrationDataStoreConfiguration> integrationDataStoreConfiguration, ILogger<AzureConnectionStringClientInitializerV2> logger)
+        {
+            EnsureArg.IsNotNull(integrationDataStoreConfiguration?.Value, nameof(integrationDataStoreConfiguration));
+            EnsureArg.IsNotNull(logger, nameof(logger));
+
+            _integrationDataStoreConfiguration = integrationDataStoreConfiguration.Value;
```

Comment:
same~ integrationDataStoreConfiguration?.Value;

Reply:
We have precheck here:
EnsureArg.IsNotNull(integrationDataStoreConfiguration?.Value, nameof(integrationDataStoreConfiguration));